src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -138,6 +139,13 @@ private static (List<Error>, TableOfContentsModel) LoadTocModel(JToken tocToken)
                 tocModelItem.TocHref = resolvedTocHref;
                 tocModelItem.Name = tocModelItem.Name ?? resolvedTopicName;
                 tocModelItem.Items = subChildren?.Items ?? tocModelItem.Items;
+
+                // validate
+                // todo: how to do required validation in strong model
+                if (tocModelItem.Name != null && string.IsNullOrEmpty(tocModelItem.Name))
```

Comment:
This check looks wierd, did you mean

```suggestion
                if (string.IsNullOrEmpty(tocModelItem.Name))
```

Reply:
we need make sure tocModelItem.Name is not null to get source info...

Reply:
I see, if I turn `SourceInfo<T>` into a struct, does it makes null handling easier?
```
struct SourceInfo<T>
{
    public T Value;
    public SourceInfo SourceInfo;
}
```

Reply:
I doubt about that, it will make the sourceInf usage complex and don't resolve our problem since the source info is still not correct(should be parent's)